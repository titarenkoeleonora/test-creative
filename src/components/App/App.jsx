import { useRef } from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";

import Header from "../Header/Header";
import Menu from "../Menu/Menu";
import Checkboxes from "../Checkboxes/Checkboxes";
import Movement from "../Movement/Movement";
import Substring from "../Substring/Substring";
import Tasks from "../Tasks/Tasks";

const App = ({animals}) => {
  const menuRef = useRef(null);
  const pageMainRef = useRef(null);

  const toggleMenu = () => {
    menuRef.current.classList.toggle("menu--toggle");
    pageMainRef.current.classList.toggle("page-main--menu-toggle");
  };

  return (
    <>
      <Router>
        <Header toggleMenu={toggleMenu} />
        <main className="page-main" ref={pageMainRef}>
          <Menu myref={menuRef} />
          <Route path="/" component={Tasks} exact />
          <Route path="/movement" component={Movement} />
          <Route path="/substring" component={Substring} />
          <Route
            path="/checkboxes"
            component={() => (<Checkboxes animals={animals} />)}
          />
        </main>
      </Router>
    </>
  );
};

export default App;
