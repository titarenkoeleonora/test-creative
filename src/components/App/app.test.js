import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

const animals = [
  {
    id: "dog",
    title: "Собака",
  },
  {
    id: "cat",
    title: "Кот",
  },
  {
    id: "mouse",
    title: "Мышь",
  },
];

it('Render App', () => {
  const tree = renderer
    .create(
      <App
        animals={animals}
      />
    );

  expect(tree).toMatchSnapshot();
});