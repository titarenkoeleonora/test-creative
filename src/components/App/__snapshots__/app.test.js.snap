// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <button
          aria-label="Показать/скрыть меню"
          className="header__menu-toggle"
          onClick={[Function]}
          type="button"
        >
          Меню
        </button>
      </div>
    </div>
  </header>,
  <main
    className="page-main"
  >
    <section
      className="menu"
    >
      <div
        className="menu__wrapper"
      >
        <nav
          className="menu__navigation"
        >
          <ul
            className="sidebar menu__sidebar"
          >
            <li
              className="sidebar__item"
            >
              <a
                className="sidebar__link"
                href="/"
                onClick={[Function]}
              >
                Главная страница
              </a>
            </li>
            <li
              className="sidebar__item"
            >
              <a
                className="sidebar__link"
                href="/movement"
                onClick={[Function]}
              >
                Функция сдвига элементов вправо на N шагов
              </a>
            </li>
            <li
              className="sidebar__item"
            >
              <a
                className="sidebar__link"
                href="/substring"
                onClick={[Function]}
              >
                Функция нахождения самой длинной последовательности 2х строк
              </a>
            </li>
            <li
              className="sidebar__item"
            >
              <a
                className="sidebar__link"
                href="/checkboxes"
                onClick={[Function]}
              >
                Отображение/скрытие элементов при переключении чек-боксов
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </section>
    <section
      className="tasks"
    >
      <div
        className="container"
      >
        <div
          className="tasks__wrapper"
        >
          <h2
            className="tasks__title"
          >
            Выполненные задания
          </h2>
          <ul
            className="tasks__list"
          >
            <li
              className="tasks__item"
            >
              <a
                className="tasks__link"
                href="/movement"
                onClick={[Function]}
              >
                Функция сдвига элементов массива вправо на N шагов.
              </a>
            </li>
            <li
              className="tasks__item"
            >
              <a
                className="tasks__link"
                href="/substring"
                onClick={[Function]}
              >
                Функция нахождения самой длинной общей последовательности 2х строк.
              </a>
            </li>
            <li
              className="tasks__item"
            >
              <a
                className="tasks__link"
                href="/checkboxes"
                onClick={[Function]}
              >
                Отображение/скрытие элементов при переключении чекбоксов.
              </a>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </main>,
]
`;
